src/sentry/db/models/manager/base.py

```diff
@@ -222,22 +166,6 @@ def __post_save(self, instance: M, **kwargs: Any) -> None:
             logger.exception(str(e))
         instance._state.db = db
 
-        # Kill off any keys which are no longer valid
-        if instance in self.__cache:
-            for key in self.cache_fields:
-                if key not in self.__cache[instance]:
-                    continue
-                value = self.__cache[instance][key]
-                current_value = self.__value_for_field(instance, key)
-                if value != current_value:
-                    cache.delete(
-                        key=self.__get_lookup_cache_key(**{key: value}), version=self.cache_version
-                    )
```

Comment:
If anyone can see what use this code is let me know... We already go through `self.cache_fields` 10 lines or so before and store the latest value of this field. So as far as I can tell, this isn't really doing anything useful. 

I think the main useful thing it does is that if a key is a non-pk and it changes, then we don't want the old key to successfully return this value anymore. That's probably a valid case... I think that it's more easily handled by rechecking the value after fetch to make sure the value we looked up by hasn't changed. I'll implement that before merge

Reply:
Actually, I might leave this in place, but figure out a way to simplify the local cache stuff...